pyproject.toml

```diff
@@ -1,20 +1,82 @@
-[tool.codespell]
-builtin = "clear,rare,informal,names,usage"
-ignore-words = "ignore_words.txt"
+[build-system]
+build-backend = "setuptools.build_meta"
+requires = [
+  "setuptools",
+  "setuptools-scm",
+]
+
+[project]
+name = "sphinx-gallery"
+description = "A Sphinx extension that builds an HTML gallery of examples from any set of Python scripts."
+readme = "README.rst"
+license = {"text" = '3-clause BSD'}
+authors = [
+   {name = "Óscar Nájera", email = 'najera.oscar@gmail.com'},
+]
+requires-python = '>=3.8'
+classifiers = [
+  'Development Status :: 4 - Beta',
+  'Framework :: Sphinx :: Extension',
+  'Intended Audience :: Developers',
+  'Programming Language :: Python',
+  "Programming Language :: Python :: 3 :: Only",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+]
+dynamic = [
+  "version",
+]
+dependencies = [
+  "pillow",
+  "sphinx>=4",
+]
+[project.optional-dependencies]
+jupyterlite= [
+  "jupyterlite_sphinx",
+]
+recommender = [
+  "numpy",
+]
+show_api_usage= [
+  "graphviz",
+]
+show_memory= [
+  "memory_profiler",
+]
+[project.urls]
+Documentation = "https://sphinx-gallery.github.io"
+Source = "https://github.com/sphinx-gallery/sphinx-gallery"
+[project.scripts]
+sphinx_gallery_py2jupyter = "sphinx_gallery.notebook:python_to_jupyter_cli"
+
+[tool.setuptools]
+include-package-data = true
+
+[tool.setuptools.dynamic]
+version = {attr = "sphinx_gallery.__version__"}
+
+[tool.setuptools.packages.find]
+where = ["."]  # list of folders that contain the packages (["."] by default)
+include = ["sphinx_gallery*"]  # package names should match these glob patterns (["*"] by default)
+exclude = ["sphinx_gallery.tests*"]  # exclude packages matching these glob patterns (empty by default)
```

Comment:
I think we should / might as well include tests

Reply:
```suggestion
```